packages/devtools/lib/src/timeline/event_details.dart

```diff
@@ -186,66 +176,132 @@ class EventDetails extends CoreElement {
   }
 }
 
-class _UiEventDetails extends CoreElement {
-  _UiEventDetails(TimelineController timelineController) : super('div') {
+class _CpuProfiler extends CoreElement {
+  _CpuProfiler(this._timelineController) : super('div') {
     layoutVertical();
     flex();
 
-    add([
-      flameChart = CpuFlameChart(timelineController),
-      bottomUp = CpuBottomUp()..attribute('hidden', true),
-      callTree = CpuCallTree()..attribute('hidden', true),
+    add(views = [
+      flameChart = CpuFlameChart(_timelineController),
+      bottomUp = CpuBottomUp(_timelineController)..hide(),
+      callTree = CpuCallTree(_timelineController)..hide(),
     ]);
   }
 
+  final TimelineController _timelineController;
+
   CpuFlameChart flameChart;
 
   CpuBottomUp bottomUp;
 
   CpuCallTree callTree;
 
-  Future<void> showTab(EventDetailsTabType tabType) async {
-    switch (tabType) {
-      case EventDetailsTabType.flameChart:
-        await flameChart.show();
-        bottomUp.attribute('hidden', true);
-        callTree.attribute('hidden', true);
-        break;
-      case EventDetailsTabType.bottomUp:
-        flameChart.attribute('hidden', true);
-        bottomUp.attribute('hidden', false);
-        callTree.attribute('hidden', true);
-        break;
-      case EventDetailsTabType.callTree:
-        flameChart.attribute('hidden', true);
-        bottomUp.attribute('hidden', true);
-        callTree.attribute('hidden', false);
-        break;
+  List<CpuProfilerView> views;
+
+  CpuProfilerViewType _selectedViewType = CpuProfilerViewType.flameChart;
+
+  bool showingMessage = false;
+
+  void showView(CpuProfilerViewType showType) {
+    _selectedViewType = showType;
+
+    // If we are showing a message, we do not want to show any other views.
+    if (showingMessage) return;
```

Comment:
this leaves _showViewType inconsistent with the show states for the views. Is that intentional? It could be noce to keep _selectedViewType consistent or make sure all views are hidden if we are showingMessage, etc.